cdap-ui/graphql/graphql.js

```diff
@@ -68,15 +62,26 @@ const resolvers = {
   },
   ProgramRecord: { __resolveType: programsTypeResolver.ProgramRecord.__resolveType },
   Workflow: {
-    runs: runsResolver.Workflow.runs,
+    runs: programsTypeResolver.Workflow.runs,
     schedules: schedulesResolver.Workflow.schedules,
   },
+  Spark: {
+    runs: programsTypeResolver.Spark.runs,
+  },
+  MapReduce: {
+    runs: programsTypeResolver.MapReduce.runs,
+  },
   ScheduleDetail: { nextRuntimes: nextRuntimesResolver.ScheduleDetail.nextRuntimes },
 };
 
 const server = new ApolloServer({
   typeDefs,
   resolvers,
+  context: ({ req }) => {
+    const auth = req.headers.authorization;
```

Comment:
check for existence of authorization first

Reply:
What if `req.headers.authorization` is `undefined`? Return `undefined`? `null`?

Reply:
depends on how the context is being used. I would say return empty object

Reply:
Yes, I am returning an empty object.